### PR TITLE
ArC: beans injected into All List injection points should be unremovable

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java [hunk 1]
```diff
@@ -466,7 +466,7 @@ public ObserverRegistrationPhaseBuildItem registerSyntheticObservers(BeanRegistr
         // Initialize the type -> bean map
         beanRegistrationPhase.getBeanProcessor().getBeanDeployment().initBeanByTypeMap();
 
-        // Register a synthetic bean for each List<?> with qualifier @All
+        // Validate @All List<> injection points
```
Comment (Ladicek) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java:469]: The code does both, though, right? The loop does validation, and then there's registration too.

**Review by Ladicek — APPROVED** [review]

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java (hunk 1 above)
Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java:469]: Ah, in fact we can remove the method completely!

**Comment by mkouba** [comment]
FTR this pull request moves the logic directly in the `arc-processor` module. Previously, we produced the `UnremovableBeanBuildItem`s in the extension. Which does not make a lot of sense anymore, since the `@All List<>` are handled by a built-in bean (https://github.com/quarkusio/quarkus/pull/32179).

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java (hunk 1 above)
Comment (Ladicek) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java:469]: If so, that sounds good to me :-)

**Review by manovotn — APPROVED** [review]

**Pull request merged**